src/OpenTelemetry.Instrumentation.AspNetCore/Implementation/HttpInListener.cs

```diff
@@ -60,10 +62,21 @@ public override void OnStartActivity(Activity activity, object payload)
                 return;
             }
 
+            // TODO: the line below once .NET ships new Activity
+            // Or do reflection now.
+            // activity.ActivityKind = ActivityKind.Server
+
             var request = context.Request;
 
             // see the spec https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-semantic-conventions.md
             var path = (request.PathBase.HasValue || request.Path.HasValue) ? (request.PathBase + request.Path).ToString() : "/";
+            if (activity != null)
```

Comment:
>activity [](start = 16, length = 8)

is it possible the Activity can be null?

Reply:
No. I didnt need the activity null check.